src/rep.rs

```diff
@@ -194,29 +201,35 @@ pub struct HostConfig {
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "PascalCase")]
-pub struct Config {
-    pub attach_stderr: bool,
-    pub attach_stdin: bool,
-    pub attach_stdout: bool,
+pub struct ContainerConfig {
+    pub hostname: Option<String>,
+    pub domainname: Option<String>,
+    pub user: Option<String>,
+    pub attach_stdin: Option<bool>,
+    pub attach_stdout: Option<bool>,
```

Comment:
I don't think those fields should be `Option`al. They seem to have default values and when the value is missing the just return either `""` or `false`.

Reply:
I don't know about other fields, but for example `args_escaped` doesn't exist, when the value is missing (I checked it) and as a result - we get `SerdeJsonError`.

Do you know, where we can find information about it? Are you sure about other fields?

Reply:
Perhaps `args_escaped` might not exist. I wanted to point out here that you should double check before making everything optional. If you're getting a `SerdeJsonError` then adjust the code accordingly and make it optional at this field.

As I pointed out in another comment the docs are very loose and the only way I know of is trial and error by first making all fields non-optional and later adjusting them one by one where it errors out. I know it's not really pleasant to do but I don't know any other way and it would be a shame to make everything an Option as accessing the fields will become quite cumbersome.

Reply:
The most important fields to check I'd say are numbers, bools and strings as those tend to have a default value most of the time. So I'd go for no Options for those and Options for Vecs/Hashmaps (but that should also be checked beforehand).

Reply:
> later adjusting them one by one where it errors out

For some images these fields are, for some they are not. Where can we find so many different images (for example, `args_escaped` may be `true`, `false`, missing for different images)?

>  it would be a shame to make everything an Option as accessing the fields will become quite cumbersome

I agree with you, but it will be much worse if the user gets `SerdeJsonError` on some image that I have not tested.

Reply:
I think there might be a way to test it if we use image `scratch`, for example:
 - create a directory `test` with a `Dockerfile` in it
 - The dockerfile has contents:
 ```
 FROM scratch
 ENV test test
 ```
  - build it
  - inspect it
  
This will give you the most barebone image that exists because it's `scratch` used to build images from `scratch` ;) So it should have nulls on all nullable fields.

Reply:
OK, perhaps an even better solution would be to look at the `go-docker` client and check which fields have `omitempty` json option or are pointers to struct which can be null.

[Link](https://github.com/docker/go-docker/blob/master/api/types/types.go)

Reply:
It looks like it actually matches the output. As you noticed `args_escaped` are optional and they have `omitempty` flag:
https://github.com/docker/go-docker/blob/4daae26030ad00e348edddff9767924ae57a3b82/api/types/container/config.go#L57

Reply:
Ok, i just noticed this repo is from 2017...

Reply:
If this doesn't work then I'm out of ideas on how to properly handle it and would probably be in favor of using Options all over the place unless anyone comes up with a better solution.

Reply:
Thank you for researching.

I fix some fields (bb04336).

It's strange, that some tools don't respect [official swagger API file](https://github.com/moby/moby/blob/c9bbc68e75f224c4490ff6e3121be50a612edc22/api/swagger.yaml), part of codebase generated from swagger file and part of code are wrote manually.

Reply:
Awesome, looks good to me :)